pkg/ccl/backupccl/restore_data_processor.go

```diff
@@ -343,22 +455,24 @@ func (rd *restoreDataProcessor) runRestoreWorkers(
 					return done, nil
 				}
 
-				sstIter, err := rd.openSSTs(ctx, entry)
-				if err != nil {
-					return done, err
-				}
-
-				summary, err := rd.processRestoreSpanEntry(ctx, kr, sstIter)
-				if err != nil {
-					return done, err
+				idx := 0
+				for idx < len(entry.Files) {
```

Comment:
I think it would be cleaner to return something like a `done` bool from openSSTs instead of comparing the idx to the num of files in the entry. the idx should be an opaque object that the caller should not use, just pass it back to openSSTs on the next call. (in that case we will probably want to rename it to something like restartFrom or maybe you'll have better ideas..)

Reply:
done